dicom_standard/extract_sops.py

```diff
@@ -0,0 +1,58 @@
+from typing import Tuple, List
+import re
+import sys
+
+from bs4 import BeautifulSoup, Tag
+
+from dicom_standard import parse_lib as pl
+from dicom_standard.extract_attributes import attribute_table_to_list
+from dicom_standard.table_utils import AttributeDictType, table_to_dict
+
+COLUMN_TITLES = ['name', 'id', 'iod']
+TABLE_ID = 'table_B.5-1'
+IOD_ABBREVIATIONS = {
+    'Computed Radiography': 'CR',
+    'Computed Tomography': 'CT',
+    'Magnetic Resonance': 'MR',
+    'Nuclear Medicine': 'NM',
+    'Ultrasound': 'US',
+    'Secondary Capture': 'SC',
+    'Radiotherapy': 'RT',
+    'Positron Emission Tomography': 'PET',
+    'Electrocardiogram': 'ECG',
+    'Electrophysiology': 'EP',
+    'OCT': 'OCT Image',
+    'Enhanced X-Ray RF Image': 'Enhanced XRF Image',
+    'X-Ray Radiofluoroscopic': 'XRF',
+}
+ID_PATTERN = re.compile(r'\b(' + '|'.join(IOD_ABBREVIATIONS.keys()) + r')\b')
```

Comment:
I don't see these abbreviations or the ID Pattern used anywhere-- is this leftover from an old iteration?

I'm not super clear on the context here, but I would caution against maintaining a list of IODs outside of the standard. This seems like something that could go stale over time and might not be obvious that we need to keep it up to date.

Reply:
Ah, I actually had removed the line that used this [here](https://github.com/innolitics/dicom-standard/pull/25/commits/fe1138f99ba04d74bda8b719838ed37770fe4219) because I had intended on getting further clarification regarding the naming convention that should be used.

Each object in the `sops.json` file contains a property `ciod` that contains the name of the associated CIOD:
![image](https://user-images.githubusercontent.com/9055029/78111840-e7524680-73c2-11ea-970c-0bcb5c1c709c.png)

However, the naming convention between the `IOD specification` column and the CIOD name within `ciods.json` isn't consistent:
![image](https://user-images.githubusercontent.com/9055029/78112005-2e403c00-73c3-11ea-9079-d75f4fd39346.png)
![image](https://user-images.githubusercontent.com/9055029/78112024-37c9a400-73c3-11ea-81c1-e006af452162.png)

Thus, the intention was to manually convert the CIOD names into a format consistent with those in `ciods.json`

Reply:
Huh, that is interesting. We may be parsing the "short" name in `ciods.json` but need to actually grab the "long" name as well. I see it specified, at least for this case, in the header for the CIOD description: http://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_A.2.html

Reply:
That makes sense regarding the mapping-- if we can avoid manually creating these mappings we should, but in the case of inconsistencies we can't resolve, it makes sense. If the above "long name parsing" strategy is not workable, then I think a mapping table makes sense.

Reply:
I had actually tried that earlier before making the mapping table: it's not consistent either haha 
![image](https://user-images.githubusercontent.com/9055029/78156636-aaa73f00-7404-11ea-84d6-0d7a0824893e.png)